tests/test_base.py

```diff
@@ -1043,3 +1050,52 @@ class TestSchema(Schema):
     assert side_effects == [1]
     TestSchema({'lazy_side_effect': 1})
     assert side_effects == [1]
+
+
+def test_uuid_field():
+    my_id = '1a4c0351-3621-4860-b528-5ada0003fbda'
+
+    with pytest.raises(ValidationError):
+        UUID().clean('x')
+
+    with pytest.raises(ValidationError):
+        assert UUID().clean(None)
+
+    assert UUID().clean(my_id) == PythonUUID(my_id)
+    assert UUID().serialize(PythonUUID(my_id)) == my_id
+
+
+def test_clean_dict():
+    f = CleanDict(key_schema=Regex(r'^k*$'), value_schema=Regex(r'^v*$'))
+    assert f.clean({'k': 'v'}) == {'k': 'v'}
+    assert f.clean({'kkk': 'vvv', 'k': 'v'}) == {'kkk': 'vvv', 'k': 'v'}
+
+    with pytest.raises(ValidationError):
+        f.clean({'a': 'v'})
+
+    with pytest.raises(ValidationError):
+        f.clean({'k': 'a'})
+
+    with pytest.raises(ValidationError):
+        f.clean({'a': 'a'})
+
+    with pytest.raises(ValidationError):
+        f.clean({'a': 1})
+
+    with pytest.raises(ValidationError):
+        f.clean({'a': [1]})
+
+    class TestSchema(Schema):
```

Comment:
This looks like it could be a separate test so that we don't test too many things in a single test. Then, if you wanna group the two CleanDict-related tests into one logical unit, put them both as methods in a test class.